cognite/neat/core/data_stores/metrics.py

```diff
@@ -0,0 +1,39 @@
+from typing import Iterable
+
+from cognite.client import CogniteClient
+from prometheus_client import REGISTRY, Counter, Gauge, Metric
+
+
+class NeatMetricsCollector:
+    def __init__(self, name: str, cdf_client: CogniteClient = None) -> None:
+        self.name = name
+        self.metrics: dict[str, Gauge | Counter] = {}
+
+    def register_metric(
+        self, metric_name: str, metric_description: str = "", m_type: str = "gauge", metric_labels: list[str] = None
```

Comment:
Suggestion: rename `m_type` to `metric_type`

Reply:
Good point , will update